resources/kubectl.rb

```diff
@@ -9,6 +11,43 @@
 end
 
 action :install do
+  platform_cmd = Mixlib::ShellOut.new('uname')
+  platform_cmd.run_command
+  platform_cmd.error!
+  platform = platform_cmd.stdout.strip.downcase
+
+  version = new_resource.version
+
+  arch_cmd = Mixlib::ShellOut.new('uname -m')
+  arch_cmd.run_command
+  arch_cmd.error!
+  arch = arch_cmd.stdout.strip
+
+  case arch
+  when 'x86', 'i686', 'i386'
+    arch = '386'
+  when 'x86_64', 'aarch64'
+    arch = 'amd64'
+  when 'armv5*'
+    arch = 'armv5'
+  when 'armv6*'
+    arch = 'armv6'
+  when 'armv7*'
+    arch = 'armv7'
+  else
+    arch = 'default'
+  end
+
+  bash 'clean up the existing kubectl' do
```

Comment:
should clean up when the installed version and installing version mismatches only, see the previous code.